### PR TITLE
Feature/paselc 525

### PR DESCRIPTION
**Pull request by GiacomoBrancazi**
#### List of Changes

add api getGecBundlesByCI

#### Motivation and Context

add api getGecBundlesByCI

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by GabrieleMaiocchiFilippo — APPROVED** [review]

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/mapper/GecMapper.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GecMapper {
+
+    public static BundleResource toResource(Bundle bundle){
+        if(bundle == null){
+            return null;
+        }
+        BundleResource response = new BundleResource();
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/mapper/GecMapper.java:22]: if the source object is identical to the destination object you can use [modelMapper](https://modelmapper.org/). It maps in automatic way two objects.

File: connector/rest/src/main/java/it/pagopa/selfcare/pagopa/backoffice/connector/rest/client/GecRestClient.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+@FeignClient(name = "${rest-client.gec.serviceCode}", url = "${rest-client.gec.base-url}")
+public interface GecRestClient extends GecConnector {
+
+    @GetMapping(value = "${rest-client.gec.getBundlesByCI.path}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    Bundles getBundlesByCI(@RequestParam(required = true) String cifiscalcode,
+                           @RequestParam(required = false) Integer limit,
+                           @RequestParam(required = false) Integer page,
+                           @RequestHeader(name = "X-Request-Id", required = false) String xRequestId);
```
Comment (jacopocarlini) [inline connector/rest/src/main/java/it/pagopa/selfcare/pagopa/backoffice/connector/rest/client/GecRestClient.java:24]: You can use a FeignConfiguration to add common headers (for ex: `requestID` or `apikey`) to every call.

File: app/src/main/resources/swagger/api-docs.json [hunk 3]
```diff
@@ -3049,6 +3052,203 @@
...
+              }
+            }
+          }
+        },
+        "security" : [ {
+          "bearerAuth" : [ "global" ]
+        } ]
+      }
+    },
+    "/gec/touchpoints" : {
```
Comment (jacopocarlini) [inline app/src/main/resources/swagger/api-docs.json:3168]: Can you change with `GET /bundles/touchpoints`.
👀  See the [issue](https://pagopa.atlassian.net/browse/PASELC-528)

File: app/src/main/resources/swagger/api-docs.json [hunk 4]
```diff
@@ -3049,6 +3052,203 @@
         } ]
       }
     },
+    "/gec/psps/bundles" : {
```
Comment (jacopocarlini) [inline app/src/main/resources/swagger/api-docs.json:3055]: change with `GET /psps/{pspCode}/bundles`

**Review by jacopocarlini — CHANGES_REQUESTED** [review]
it's ok, just some observations.
PS: check some code smells for sonar please

File: app/src/main/resources/swagger/api-docs.json (hunk 4 above)
Comment (GiacomoBrancazi) [reply app/src/main/resources/swagger/api-docs.json]: fix

File: app/src/main/resources/swagger/api-docs.json (hunk 3 above)
Comment (GiacomoBrancazi) [reply app/src/main/resources/swagger/api-docs.json]: fix

File: openapi/openapi.json [hunk 5]
```diff
@@ -28,6 +28,10 @@
       "name": "creditor-institutions",
       "description": "Api Config creditor institution's operations"
     },
+    {
+      "name": "gec",
```
Comment (jacopocarlini) [inline openapi/openapi.json:32]:
```suggestion
      "name": "Bundles",
```

File: openapi/openapi.json [hunk 6]
```diff
@@ -28,6 +28,10 @@
...
+      "description": "Gec Controller"
```
Comment (jacopocarlini) [inline openapi/openapi.json:33]:
```suggestion
      "description": "All about commissional Bundles"
```

File: openapi/openapi.json [hunk 7]
```diff
@@ -3379,6 +3383,228 @@
         ]
       }
     },
+    "/gec/bundles/touchpoints": {
```
Comment (jacopocarlini) [inline openapi/openapi.json:3386]:
```suggestion
    "/bundles/touchpoints": {
```

File: openapi/openapi.json [hunk 8]
```diff
@@ -3379,6 +3383,228 @@
...
+            "description": "Number of elements on one page. Default = 50",
+            "required": false,
+            "style": "form",
+            "schema": {
+              "type": "integer",
+              "format": "int32"
+            }
+          },
+          {
+            "name": "page",
```
Comment (jacopocarlini) [inline openapi/openapi.json:3407]: add default `0`

File: openapi/openapi.json [hunk 9]
```diff
@@ -3379,6 +3383,228 @@
...
+        "tags": [
+          "gec"
+        ],
+        "summary": "getTouchpoints",
+        "description": "Get paginated list of bundles",
+        "operationId": "getTouchpointsUsingGET",
+        "parameters": [
+          {
+            "name": "limit",
+            "in": "query",
```
Comment (jacopocarlini) [inline openapi/openapi.json:3397]: add default `10`

File: openapi/openapi.json [hunk 10]
```diff
@@ -3379,6 +3383,228 @@
...
+        "security": [
+          {
+            "bearerAuth": [
+              "global"
+            ]
+          }
+        ]
+      }
+    },
+    "/gec/{pspCode}/bundles": {
```
Comment (jacopocarlini) [inline openapi/openapi.json:3479]:
```suggestion
    "/psp/{pspCode}/bundles": {
```

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/gec/BundleResource.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    private Integer maxPaymentAmount;
+
+    @JsonProperty("payment_type")
+    private String paymentType;
+
+    @JsonProperty("touchpoint")
+    private String touchpoint;
+
+    @JsonProperty("type")
+    private String type;
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/gec/BundleResource.java:40]: change with an enum [private, global, public]

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/gec/BundleResource.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    @JsonProperty("id_psp")
+    private String idPsp;
+
+    @JsonProperty("name")
+    private String name;
+
+    @JsonProperty("description")
+    private String description;
+
+    @JsonProperty("payment_amount")
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/gec/BundleResource.java:24]:
```suggestion
    @JsonProperty("payment_amount")
    @Schema(description = "The fees of this bundle")
```

File: connector/rest/src/main/java/it/pagopa/selfcare/pagopa/backoffice/connector/rest/client/GecRestClient.java (hunk 2 above)
Comment (GiacomoBrancazi) [reply connector/rest/src/main/java/it/pagopa/selfcare/pagopa/backoffice/connector/rest/client/GecRestClient.java:24]: we can use FeignConfiguration in future. It's an impactful change

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/mapper/GecMapper.java (hunk 1 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/mapper/GecMapper.java:22]: ok thanks for the new option I will try to implement it in the next development

**Pull request merged**